File: library/api/src/main/java/org/cru/godtools/api/ViewsApi.java
package org.cru.godtools.api;

import org.ccci.gto.android.common.jsonapi.model.JsonApiObject;
import org.cru.godtools.api.model.ToolViews;

import androidx.annotation.NonNull;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface ViewsApi {
    String PATH_VIEWS = "views";

    @POST(PATH_VIEWS)
    Call<JsonApiObject<ToolViews>> submitViews(@Body @NonNull ToolViews views);
}

File: library/api/src/main/java/org/cru/godtools/api/ViewsApi.kt
package org.cru.godtools.api

import okhttp3.ResponseBody
import org.cru.godtools.api.model.ToolViews
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST

private const val PATH_VIEWS = "views"

interface ViewsApi {
    @POST(PATH_VIEWS)
    suspend fun submitViews(@Body views: ToolViews): Response<ResponseBody>
}
